internal/plugin/connectors/tcp/proxy_service.go

```diff
@@ -110,6 +110,9 @@ func (proxy *proxyService) handleConnection(clientConn net.Conn) error {
 	logger.Infof("New connection on %v.\n", clientConn.LocalAddr())
 
 	targetConn, err = proxy.connector.Connect(clientConn, backendCredentials)
+	if err == io.EOF {
```

Comment:
I'd like to see a comment explaining why we need to handle these two cases differently and (perhaps) why you chose the word "care".